docs/build/guides/transactions/clawbacks.mdx

```diff
@@ -44,354 +44,865 @@ Here we’ll cover the following approaches to clawing back an asset.
 
-### Preamble: Issuing a Clawback-able Asset
+### Preamble: Creating + Funding Accounts and Issuing a Clawback-able Asset
 
 First, we’ll set up an account to enable clawbacks and issue an asset accordingly.
 
 Properly issuing an asset (with separate issuing and distribution accounts) is a little more involved, but we’ll use a simpler method here.
 
 Also, note that we first need to enable clawbacks and then establish trustlines since you cannot retroactively enable clawback on existing trustlines.
 
+:::info
+
+The following code snippet contains helper functions that will be used in the following examples
+
+:::
+
 <CodeExample>
 
 ```js
-const sdk = require("stellar-sdk");
+import * as sdk from "@stellar/stellar-sdk";
+
+let server = new sdk.rpc.Server("https://soroban-testnet.stellar.org");
 
-let server = new sdk.Server("https://horizon-testnet.stellar.org");
+const A = sdk.Keypair.random();
+const B = sdk.Keypair.random();
+const C = sdk.Keypair.random();
 
-const A = sdk.Keypair.fromSecret(
-  "SAQLZCQA6AYUXK6JSKVPJ2MZ5K5IIABJOEQIG4RVBHX4PG2KMRKWXCHJ",
-);
-const B = sdk.Keypair.fromSecret(
-  "SAAY2H7SANIS3JLFBFPLJRTYNLUYH4UTROIKRVFI4FEYV4LDW5Y7HDZ4",
-);
-const C = sdk.Keypair.fromSecret(
-  "SCZANGBA5YHTNYVVV4C3U252E2B6P6F5T3U6MM63WBSBZATAQI3EBTQ4",
-);
+console.log("=== ACCOUNT SETUP ===");
+console.log(`Account A (Issuer): ${A.publicKey()}`);
+console.log(`Account B (Trustor): ${B.publicKey()}`);
+console.log(`Account C (Trustor): ${C.publicKey()}`);
+console.log();
 
 const ASSET = new sdk.Asset("CLAW", A.publicKey());
 
-/// Enables AuthClawbackEnabledFlag on an account.
+// Helper function to format account ID with label
+function formatAccount(accountId) {
+  const shortId = accountId.substring(0, 8);
+  if (accountId === A.publicKey()) {
+    return `${shortId} (Account A)`;
+  } else if (accountId === B.publicKey()) {
+    return `${shortId} (Account B)`;
+  } else if (accountId === C.publicKey()) {
+    return `${shortId} (Account C)`;
+  }
+  return shortId;
+}
+
+// Helper function to fetch claimable balance details
+async function fetchClaimableBalance(
+  balanceId,
+  description = "Claimable Balance",
+) {
+  try {
+    console.log(`\n--- Checking ${description} ---`);
+    console.log(`Looking up balance ID: ${balanceId}`);
+
+    // Parse the claimable balance ID from hex XDR
+    const claimableBalanceId = sdk.xdr.ClaimableBalanceId.fromXDR(
+      balanceId,
+      "hex",
+    );
+
+    // Create ledger key for the claimable balance entry
+    const ledgerKey = sdk.xdr.LedgerKey.claimableBalance(
+      new sdk.xdr.LedgerKeyClaimableBalance({
+        balanceId: claimableBalanceId,
+      }),
+    );
+
+    // Use SDK's getLedgerEntries method with XDR object array
+    const response = await server.getLedgerEntries(ledgerKey);
```

Comment:
You can use the new one https://github.com/stellar/js-stellar-sdk/pull/1221 for this: `server.getClaimableBalance(balanceId)`, even shorter!